### PR TITLE
Java solid kata solution

### PR DESCRIPTION
**Pull request by BAIRU-SHASHI-KANTH**
https://kata-log.rocks/task-list-kata
### Features Implemented : 

**Deadlines**
Give each task an optional deadline with the **deadline ID date** command.
Show all tasks due today with the **today** command.

**Deletion**
Allow users to delete tasks with the **delete ID** command.

**Views**
View tasks by deadline with the **view by deadline** command.
Don’t remove the functionality that allows users to view tasks by project, but change the command to **view by project**.

### CODE REVIEW

File: java/src/main/java/com/codurance/training/tasks/ExecuteCommands.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                break;
+            case "delete":
+                new DeleteTask().delete(commandRest[1],taskList.out,taskList.tasks);
+                break;
+            case "help":
+                new Help().help(taskList.out);
+                break;
+            default:
+                new Error().error(command,taskList.out);
+                break;
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/ExecuteCommands.java]: should not create instances for each execution call

File: java/src/main/java/com/codurance/training/tasks/AddProject.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.codurance.training.tasks;
+
+import java.util.ArrayList;
+public class AddProject {
+
+    public void add(String name,TaskList taskList) {
+        taskList.tasks.put(name, new ArrayList<Task>());
+    }
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/AddProject.java]: add interface and have taskList as class field and maintain a constructor

File: java/src/main/java/com/codurance/training/tasks/AddTask.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.codurance.training.tasks;
+
+import java.util.List;
+
+public class AddTask {
+    public void add(String project, String description, TaskList taskList) {
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/AddTask.java]: add interface and have taskList as class field and maintain a constructor

File: java/src/main/java/com/codurance/training/tasks/TaskList.java [hunk 4]
```diff
@@ -4,26 +4,23 @@
...
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Calendar;
+
 public final class TaskList implements Runnable {
     private static final String QUIT = "quit";
 
-    private final Map<String, List<Task>> tasks = new LinkedHashMap<>();
+    public final Map<String, List<Task>> tasks = new LinkedHashMap<>();
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/TaskList.java]: this should not be public, we should encapsulate our code

File: java/src/main/java/com/codurance/training/tasks/TaskList.java [hunk 5]
```diff
@@ -4,26 +4,23 @@
...
     private final BufferedReader in;
-    private final PrintWriter out;
-
+    public final PrintWriter out;
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/TaskList.java]: this should not be public, we should encapsulate our code

File: java/src/main/java/com/codurance/training/tasks/DeleteTask.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package com.codurance.training.tasks;
+
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+
+public class DeleteTask {
+    public void delete(String idString, PrintWriter out, Map<String, List<Task>> tasks) {
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/DeleteTask.java]: add interface and have taskList as class field and maintain a constructor

File: java/src/main/java/com/codurance/training/tasks/ShowFactory.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        Show show = null;
+
+        switch (commandRest){
+            case "by project":
+                show = new ShowTasks();
+                break;
+            case "by deadline":
+                show = new ShowTaskWithDeadlines();
+                break;
+            default:
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/ShowFactory.java]: should not create instances for each call

File: java/src/main/java/com/codurance/training/tasks/ShowTaskWithDeadlines.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
+package com.codurance.training.tasks;
+
+
+import java.io.PrintWriter;
+        import java.util.List;
+        import java.util.Map;
+
+public class ShowTaskWithDeadlines implements Show{
+    @Override
+    public void show(PrintWriter out, Map<String, List<Task>> tasks) {
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/ShowTaskWithDeadlines.java]: have tasks as class field and maintain a constructor

File: java/src/main/java/com/codurance/training/tasks/ShowTasks.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
+package com.codurance.training.tasks;
+
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+
+public class ShowTasks implements Show{
+    @Override
+    public void show(PrintWriter out, Map<String, List<Task>> tasks) {
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/ShowTasks.java]: have tasks as class field and maintain a constructor

File: java/src/main/java/com/codurance/training/tasks/ShowTodayDeadlineTasks.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public class ShowTodayDeadlineTasks implements Show{
+
+    public void show(PrintWriter out, Map<String, List<Task>> tasks) {
+        Date date = Calendar.getInstance().getTime();
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/ShowTodayDeadlineTasks.java]: have tasks as class field and maintain a constructor

File: java/src/main/java/com/codurance/training/tasks/TaskDeadline.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package com.codurance.training.tasks;
+
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+
+public class TaskDeadline {
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/TaskDeadline.java]: add interface and have tasks as class field and maintain a constructor

File: java/src/main/java/com/codurance/training/tasks/TaskList.java [hunk 12]
```diff
@@ -47,103 +44,32 @@ public void run() {
...
-            for (Task task : project.getValue()) {
-                out.printf("    [%c] %d: %s%n", (task.isDone() ? 'x' : ' '), task.getId(), task.getDescription());
-            }
-            out.println();
-        }
-    }
 
-    private void add(String commandLine) {
+
+    public void add(String commandLine) {
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/TaskList.java]: have add method in different class

File: java/src/main/java/com/codurance/training/tasks/UpdateTask.java [hunk 13]
```diff
@@ -0,0 +1,21 @@
+package com.codurance.training.tasks;
+
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+
+public class UpdateTask {
+    public void setDone(String idString, boolean done, PrintWriter out,Map<String, List<Task>> tasks) {
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/UpdateTask.java]: add interface and have tasks as class field and maintain a constructor

**Review by AbhilashZemoso — CHANGES_REQUESTED** [review]
work on comments
make sure all OOPS principles are also followed

**Comment by BAIRU-SHASHI-KANTH** [comment]
> work on comments make sure all OOPS principles are also followed

implemented the requested changes

**Review by AbhilashZemoso — APPROVED** [review]
LGTM

**Pull request merged**